isbn_updater/src/comp.php

```diff
@@ -0,0 +1,352 @@
+
+echo '<pre>';
+if (move_uploaded_file($_FILES['userfile']['tmp_name'], $uploadfile)) {
+	echo "File is valid, and was successfully uploaded.\n";
+} else {
+	echo "File is invalid, and failed to upload - Please try again. -\n";
+}
+echo "</br>";
+print_r($uploadfile);
+echo "</br>";
+echo "</br>";
+
+/**
+ * Get the user config file. This script will fail disgracefully if it has not been created and nothing will happen.
+ */
+require('../../user.config.php');
+
+echo "Tenancy Shortcode set: " . $shortCode;
+echo "</br>";
+
+echo "Client ID set: " . $clientID;
+echo "</br>";
+
+echo "User GUID to use: " . $TalisGUID;
+echo "</br>";
+
+
+//**********CREATE LOG FILE TO WRITE OUTPUT*
+
+$myfile = fopen("../../report_files/isbn_updater_output.log", "a") or die("Unable to open isbn_updater_output.log");
+fwrite($myfile, "Started | Input File: $uploadfile | Date: " . date('d-m-Y H:i:s') . "\r\n\r\n");
+fwrite($myfile, "Item ID" . "\t" . "Old ISBN" . "\t" . "New ISBN" . "\t" . "Resource ID" . "\t" . "Update Status?" . "\r\n");
+
+
+function getToken($clientID, $secret) {
+	$tokenURL = 'https://users.talis.com/oauth/tokens';
+	$content = "grant_type=client_credentials";
+
+	//*********GET DATE**********************
+
+	$date = date('Y-m-d\TH:i:s');
+	// $date1 = "2015-12-21T15:44:36";
+
+	//************GET_TOKEN***************
+
+	$ch = curl_init();
+
+	curl_setopt($ch, CURLOPT_URL, $tokenURL);
+	curl_setopt($ch, CURLOPT_POST, TRUE);
+	curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+	curl_setopt($ch, CURLOPT_USERPWD, "$clientID:$secret");
+	curl_setopt($ch, CURLOPT_POSTFIELDS, $content);
+
+	$return = curl_exec($ch);
+	$info = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+	if ($info !== 200){
+		echo "<p>ERROR: There was an error getting a token:</p><pre>" . var_export($return, true) . "</pre>";
+	} else {
+		echo "Got Token</br>";
+	}
+
+	curl_close($ch);
+
+	$jsontoken = json_decode($return);
+
+	if (!empty($jsontoken->access_token)){
+		$token = $jsontoken->access_token;
+	} else {
+		echo "<p>ERROR: Unable to get an access token</p>";
+		exit;
+	}
+
+	return $token;
+}
+
+/**
+ * isValidIsbn13
+ * https://en.wikipedia.org/wiki/ISBN#ISBN-13_check_digit_calculation
+ *
+ * @param  string $isbn
+ * @return boolean
+ */
+function isValidIsbn13($isbn)
+{
+    $check = 0;
+
+	// if it looks like an ISBN13
+	if (preg_match("/^97[89]\d{9}[\dxX]$/", $isbn)){
+		for ($i = 0; $i < 13; $i += 2) {
+			$check += (int)$isbn[$i];
+		}
+		
+		for ($i = 1; $i < 12; $i += 2) {
+			$check += 3 * $isbn[$i];
+		}
+		// and if the check digit is valid.
+		return (0 === ($check % 10)) ? true : false;
+	}
+
+	// else it can't be an ISBN13
+	return false;		
+}
+
+/**
+ * Update the resource if we need to
+ *
+ * @param  mixed $resource
+ * @param  mixed $old_isbn
+ * @param  mixed $new_isbn
+ * @return boolean|array False or the new ISBN13s to update
+ */
+function apply_update_logic($resource, $old_isbn, $new_isbn){
```

Comment:
Function name intent:

`apply_update_logic` makes me feel like this is going to update the resource with the new ISBNs.  Instead it is returning an array of what the final set of ISBN13s should look like.

Suggestion: rename to `newISBNArr` (or something similar).  Most other functions here are camelCase so I'm following that convention with the suggestion